js/app/widgets/DateTimeRangePicker.js

```diff
@@ -0,0 +1,191 @@
+import React, { useState, useEffect } from 'react'
+import { render } from 'react-dom'
+import moment from 'moment'
+import { ConfigProvider, DatePicker, Select } from 'antd'
+
+import 'antd/es/input/style/index.css'
+
+import { antdLocale } from '../i18n'
+
+const { Option } = Select
+
+function generateTimeSlots(disabled = false) {
+  const items = []
+  new Array(24).fill().forEach((_, index) => {
+    items.push({
+      time: moment({ hour: index }),
+      disabled,
+    })
+    items.push({
+      time: moment({ hour: index, minute: 15 }),
+      disabled,
+    })
+    items.push({
+      time: moment({ hour: index, minute: 30 }),
+      disabled,
+    })
+    items.push({
+      time: moment({ hour: index, minute: 45 }),
+      disabled,
+    })
+  })
+  return items
+}
+
+const DateTimeRangePicker = ({ defaultValue, onChange, format }) => {
+  const [values, setValues] = useState(() =>
+    defaultValue
+      ? [moment(defaultValue.after), moment(defaultValue.before)]
+      : [],
+  )
+
+  const firstSelectOptions = generateTimeSlots()
+  const [secondSelectOptions, setSecondSelectOptions] =
+    useState(generateTimeSlots())
+
+  const [timeValues, setTimeValues] = useState(
+    defaultValue
+      ? {
+          after: moment(defaultValue.after).format('HH:mm'),
+          before: moment(defaultValue.before).format('HH:mm'),
+        }
+      : {},
+  )
+
+  useEffect(() => {
+    console.log('timeValues.after dans useEffect :', timeValues.after)
+
+    const after = moment(timeValues.after, 'HH:mm')
+
+    const before = after.clone().add(15, 'minutes')
+    setTimeValues(prevState => ({
+      ...prevState,
+      before: before.format('HH:mm'),
+    }))
+    console.log('before', before)
+
+    const updatedSecondOptions = secondSelectOptions.map(option => {
+      const isBefore = after.isBefore(option.time)
+      return {
+        ...option,
+        disabled: !isBefore,
+      }
+    })
+
+    setSecondSelectOptions(updatedSecondOptions)
+  }, [timeValues.after])
```

Comment:
Okay, this definitely works, BUT (it's a bit nitpicking) why not perform this computation in `handleChange()`?
I see two different patterns/control flows to update the state here. This is okay in a small component like this, but it could lead to hidden behaviors in larger codebases.